bwscanner/circuit.py

```diff
@@ -31,7 +31,7 @@ def random_path_to_exit(exit_relay, relays):
 class CircuitGenerator(object):
     def __init__(self, state):
         self.state = state
-        self.relays = list(set(state.routers.values()))
+        self.relays = list(set(r for r in state.routers.values() if r))
```

Comment:
probably should add a comment why we are doing this (excluding None entries - why are they there?)